packages/wallets/src/evm/connectors/smart-wallet/lib/erc4337-provider.ts

```diff
@@ -134,7 +123,13 @@ export class ERC4337EthersProvider extends providers.BaseProvider {
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         confirmations?: number,
       ): Promise<providers.TransactionReceipt> => {
-        const transactionReceipt = await waitForUserOp();
+        const transactionReceipt =
+          await new Promise<providers.TransactionReceipt>((resolve, reject) => {
```

Comment:
Do you need to wrap in a promise here?

Reply:
Without it I was getting some issues, though that was before the desync fix, will test without wrapping and get back.